sdks/python/apache_beam/metrics/metric.py

```diff
@@ -293,3 +339,49 @@ def with_steps(self, steps):
 
     self._steps.update(steps)
     return self
+
+
+class MetricLogger(object):
```

Comment:
Can you test that this class works if you change the current metric environment. We should make sure that all instances of the metic on all MetricEnvironments will be logged. AS the DelegatingMetric's value changes based on the MetricEnvironment it is set in. I suspect what you are doing works fine, as long as long as you keep a separate timer for each MetricEnvironment. I.e. imagine two transforms with a metirc in two separate MetricEnvironments. If they are sharing the same time in the same MetricLogger, it may log for one metric environment, but not the other one, when it tries to log.

Note:
- Every time a new bundle is processed, a new MetricEnvironment is created, which will clear the metrics.
- Separate transforms, or code running in separate threads will store metrics in separate MetricEnvironments
- DelegatingMetrics always use the current MetricEnvironment to store a metric, or get a metric value (This is a thread local variable which is reset when a new bundle is processed, new transform is executing, etc.).

Reply:
Do you know the behaviour here of your change? Like I said, it looks like you will only be logging the state of a metric in the current MetricContainer. Rather then all of the metrics in separate MetricContainers aggregated together into a single value.

Is there a guarantee that you will log once for every metric container? (I don;'t think so because of the timing logic)

Reply:
`MetricLogger` stores separate metric values from `MetricEnvironment` for reporting a single aggregated result to the logger. In this PR, it's used for keeping process-wide API latency histogram values and periodically logging it. So I think changing `MetricEnvironment` doesn't have any effect here.

Reply:
Oh okay thanks, this clarifies it